src/com/reco1l/osu/ui/Preference.kt

```diff
@@ -0,0 +1,308 @@
+package com.reco1l.osu.ui
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.content.res.TypedArray
+import android.os.Parcel
+import android.os.Parcelable
+import android.text.TextUtils
+import android.util.AttributeSet
+import android.view.KeyEvent
+import android.view.inputmethod.EditorInfo.IME_ACTION_DONE
+import android.widget.EditText
+import androidx.core.content.res.TypedArrayUtils
+import androidx.preference.Preference
+import androidx.preference.PreferenceViewHolder
+import androidx.preference.R
+import com.reco1l.toolkt.android.hideKeyboard
+
+
+/// Select
+
+@SuppressLint("RestrictedApi")
+open class SelectPreference(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int): Preference(context, attrs, defStyleAttr, defStyleRes) {
+
+    constructor(context: Context) : this(context, null)
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : this(context, attrs, defStyleAttr, 0)
+    constructor(context: Context, attrs: AttributeSet?) : this(context, attrs, TypedArrayUtils.getAttr(context, R.attr.dialogPreferenceStyle, android.R.attr.dialogPreferenceStyle))
+
+
+    /**
+     * The selected value.
+     */
+    var value: String? = null
+        set(value) {
+
+            val changed = !field.contentEquals(value)
+
+            if (changed || !isValueSet) {
+
+                field = value
+                isValueSet = true
+
+                persistString(value)
+                dialog.setSelected(value)
+
+                if (changed) {
+                    onPreferenceChangeListener?.onPreferenceChange(this, value)
+                    notifyChanged()
+                }
+            }
+        }
+
+    /**
+     * The options to be displayed in the dialog.
+     */
+    var options: MutableList<Option>
+        get() = dialog.options
+        set(value) {
+            dialog.setOptions(value)
+        }
+
+
+    private var isValueSet = false
+
+
+    private val dialog = SelectDialog()
+
+
+    init {
+
+        val attributes = context.obtainStyledAttributes(attrs, R.styleable.ListPreference, defStyleAttr, defStyleRes)
+
+        val entries = TypedArrayUtils.getTextArray(attributes, R.styleable.ListPreference_entries, R.styleable.ListPreference_android_entries)
+        val values = TypedArrayUtils.getTextArray(attributes, R.styleable.ListPreference_entryValues, R.styleable.ListPreference_android_entryValues)
```

Comment:
As is the case with the previous comment, I have checked what `TypedArrayUtils.getTextArray()` does. Couldn't this be replaced by a simple Elvis operator?

```suggestion
        val entries = attributes.getTextArray(R.styleable.ListPreference_entries) ?: attributes.getTextArray(R.styleable.ListPreference_android_entries)
        val values = attributes.getTextArray(R.styleable.ListPreference_entryValues) ?: attributes.getTextArray(R.styleable.ListPreference_android_entryValues)
```